inc/Location.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef LOCATION_HPP
+#define LOCATION_HPP
+
+#include <iostream>
+#include <map>
+#include <vector>
+
+class Location {
+public:
+    Location();
+    ~Location();
+
+private:
+    std::string m_path;
+    std::string m_root;
+    std::string m_index;
+    std::string m_cgi_ext;
+    std::string m_cgi_path;
+    bool m_autoindex;
+    std::vector<std::string> m_allowed_methods;
```

Comment:
I don't think we need to store the allowed methods as strings. 
Since there are only a small set of methods we could define them in an enum and then use a bool array or bit field or whatever to simply check.

For example (not tested :D )

> define enum

`enum METHOD { METHOD_GET, METHOD_POST, METHOD_DELETE, METHOD_COUNT };`

> in the class we can use count for our array

`bool m_allowed_methods[METHOD_COUNT];`

> then in code check if the method is allowed

`if (m_allowed_methods[METHOD_GET]) { do stuff };`